app/controllers/notifications_controller.rb

```diff
@@ -3,5 +3,44 @@ class NotificationsController < ApplicationController
 
   def index
     @notifications = current_user.profile.notifications.order(created_at: :desc)
+    @notifications.map { |n| n.seen! if n.unseen? }
+  end
+
+  def update
+    @notification = Notification.find(params[:id])
+    @notification.clicked!
+    redirect_to_notification
+  end
+
+  private
+
+  def redirect_to_notification
+    return redirect_to_invitation if @notification.notifiable.is_a? Invitation
+    return redirect_to_comment if @notification.notifiable.is_a? Comment
+    return redirect_to_connection if @notification.notifiable.is_a? Connection
+    return redirect_to_post if @notification.notifiable.is_a? Post
+
+    redirect_to_like if @notification.notifiable.is_a? Like
```

Comment:
Aqui vale o mesmo princípio de OOP do Likes Controller, onde criamos para cada Model o seu respectivo controller, como Invitation::NotificationsController, e cada um terá sua action update. e dentro dela será algo como:
```suggestion
  def update
    super
    redirect_to_notification
  end
```

e no update do NotificationsController ficará apenas:

```rb
def update
  @notification = Notification.find(params[:id])
  @notification.clicked!
end
```

Reply:
Que da hora, Dan